compiler/IREmitter/Payload/codegen-payload.c

```diff
@@ -704,6 +704,25 @@ VALUE sorbet_int_hash_to_h(VALUE recv, ID fun, int argc, const VALUE *const rest
     return sorbet_rb_hash_to_h(recv);
 }
 
+SORBET_INLINE
+VALUE sorbet_rb_hash_fetch_m_withBlock(VALUE recv, ID fun, int argc, const VALUE *const restrict argv, BlockFFIType blk,
+                                       const struct rb_captured_block *captured, VALUE closure, int numPositionalArgs) {
+    extern VALUE sorbet_hash_stlike_lookup(VALUE hash, VALUE key, VALUE * val);
+
+    rb_check_arity(argc, 1, 2);
+    VALUE key = argv[0];
+    VALUE val;
+
+    if (sorbet_hash_stlike_lookup(recv, key, &val)) {
```

Comment:
`hash_stlike_lookup` is marked `inline` in the interpreter, right? It might be good to dash off a quick benchmark to make sure the lack of inlinability isn't hurting us here.

Reply:
My intuition was that we should be fine either way, as we'd have to call into the vm in either case to do the lookup. I'll add a benchmark just to confirm that.

Reply:
Makes sense! I didn't actually have an alternative in mind anyway. :)